### PR TITLE
Remove owner check in native app logic

### PR DESCRIPTION
**Pull request by sfc-gh-bdufour**
### Pre-review checklist
   * [X] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [X] I've added or updated automated unit tests to verify correctness of my new code.
   * [] I've added or updated integration tests to verify correctness of my new code.
   * [X] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [X] I've confirmed that my changes are up-to-date with the target branch.
   * [X] I've described my changes in the release notes.
   * [X] I've described my changes in the section below.

### Changes description

Remove ownership checks in snow app command logic. This logic is too restrictive for some advanced use cases. We should rely on RBAC for privileges rather than enforcing a stricter permission model. Fixes SNOW-1652272. 

### Testing

- Updated the current unit tests to reflect the new behaviour
- Manually verified correct and incorrect privileges by executing the CLI and forcing application upgrades as well as version drops with a different user.
- I tried writing integ tests to cover the behaviour change, but it turned out to be more complex than I expected to set them up correctly (and clean up). I decided to post the PR without integ tests at least for now, and try to backfill them in a subsequent change so that this bug fix could make the next release.

### CODE REVIEW

File: src/snowflake/cli/_plugins/nativeapp/application_entity.py [hunk 1]
```diff
@@ -221,9 +220,6 @@ def drop(
             )
             return
 
-        # 2. Check for the right owner
-        ensure_correct_owner(row=show_obj_row, role=app_role, obj_name=app_name)
-
         # 3. Check if created by the Snowflake CLI
```
Comment (sfc-gh-cgorrie) [inline src/snowflake/cli/_plugins/nativeapp/application_entity.py:223]: Numbered comment is wrong now

File: src/snowflake/cli/_plugins/nativeapp/application_package_entity.py [hunk 2]
```diff
@@ -449,11 +448,6 @@ def create_app_package(
         )
 
         if show_obj_row:
-            # 1. Check for the right owner role
-            ensure_correct_owner(
-                row=show_obj_row, role=package_role, obj_name=package_name
-            )
-
             # 2. Check distribution of the existing application package
```
Comment (sfc-gh-cgorrie) [inline src/snowflake/cli/_plugins/nativeapp/application_package_entity.py:451]: Numbered comment is wrong now

File: src/snowflake/cli/_plugins/nativeapp/application_entity.py (hunk 1 above)
Comment (sfc-gh-bdufour) [reply src/snowflake/cli/_plugins/nativeapp/application_entity.py]: fixed

File: src/snowflake/cli/_plugins/nativeapp/application_package_entity.py (hunk 2 above)
Comment (sfc-gh-bdufour) [reply src/snowflake/cli/_plugins/nativeapp/application_package_entity.py:451]: fixed

**Pull request merged**